scipy/stats/_stats_py.py

```diff
@@ -9244,41 +9244,47 @@ def combine_pvalues(pvalues, method='fisher', weights=None):
         NaN = _get_nan(pvalues)
         return SignificanceResult(NaN, NaN)
 
     if method == 'fisher':
-        statistic = -2 * np.sum(np.log(pvalues))
-        chi2 = _SimpleChi2(2 * len(pvalues))
+        statistic = -2 * xp.sum(xp.log(pvalues))
+        chi2 = _SimpleChi2(xp.asarray(2 * xp_size(pvalues), dtype=pvalues.dtype))
         pval = _get_pvalue(statistic, chi2, alternative='greater',
-                           symmetric=False, xp=np)
+                           symmetric=False, xp=xp)
     elif method == 'pearson':
-        statistic = 2 * np.sum(np.log1p(-pvalues))
-        chi2 = _SimpleChi2(2 * len(pvalues))
+        statistic = 2 * xp.sum(xp.log1p(-pvalues))
+        chi2 = _SimpleChi2(xp.asarray(2 * xp_size(pvalues), dtype=pvalues.dtype))
         pval = _get_pvalue(-statistic, chi2, alternative='less',
-                           symmetric=False, xp=np)
+                           symmetric=False, xp=xp)
     elif method == 'mudholkar_george':
-        normalizing_factor = np.sqrt(3/len(pvalues))/np.pi
-        statistic = -np.sum(np.log(pvalues)) + np.sum(np.log1p(-pvalues))
-        nu = 5 * len(pvalues) + 4
-        approx_factor = np.sqrt(nu / (nu - 2))
-        pval = distributions.t.sf(statistic * normalizing_factor
-                                  * approx_factor, nu)
+        normalizing_factor = math.sqrt(3/xp_size(pvalues))/xp.pi
+        statistic = -xp.sum(xp.log(pvalues)) + xp.sum(xp.log1p(-pvalues))
+        nu = xp.asarray(5 * xp_size(pvalues) + 4, dtype=pvalues.dtype)
+        approx_factor = math.sqrt(nu / (nu - 2))
+        t = _SimpleStudentT(nu)
+        pval = _get_pvalue(statistic * normalizing_factor * approx_factor, t,
+                           alternative="greater", symmetric=False, xp=xp)
     elif method == 'tippett':
-        statistic = np.min(pvalues)
-        beta = _SimpleBeta(1, len(pvalues))
+        statistic = xp.min(pvalues)
+        beta = _SimpleBeta(xp.asarray(1.),
+                           xp.asarray(xp_size(pvalues), dtype=statistic.dtype))
         pval = _get_pvalue(statistic, beta, alternative='less',
-                           symmetric=False, xp=np)
+                           symmetric=False, xp=xp)
     elif method == 'stouffer':
         if weights is None:
-            weights = np.ones_like(pvalues)
-        elif len(weights) != len(pvalues):
+            weights = xp.ones_like(pvalues, dtype=pvalues.dtype)
+        elif xp_size(weights) != xp_size(pvalues):
             raise ValueError("pvalues and weights must be of the same size.")
 
-        Zi = distributions.norm.isf(pvalues)
-        statistic = np.dot(weights, Zi) / np.linalg.norm(weights)
-        pval = distributions.norm.sf(statistic)
+        norm = _SimpleNormal()
+        Zi = norm.isf(pvalues)
+        statistic = weights @ Zi / xp.linalg.vector_norm(weights)
+        pval = _get_pvalue(statistic, norm, alternative="greater", symmetric=True,
+                           xp=xp)
```

Comment:
```suggestion
        pval = _get_pvalue(statistic, norm, alternative="greater", xp=xp)
```
I think it's fine to rely on the default if it saves a line. It's almost always used, and I don't think it's ever used in a way that changes the result. (It only matters if `alternative='two-sided'`.)